File: solutions/Question 1512/Solution2.py
# Problem: 1512. Number of Good Pairs
# Difficulty: Easy
# Topics: Array, Hash Table, Math, Counting
# URL: https://leetcode.com/problems/number-of-good-pairs/
# Runtime: 3 ms
# Memory: 17.9 MB
# Solved via CodePath Auto-Committer

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        c = 0
        for a in range(len(nums)):
            for b in range(a+1,len(nums)):
                if nums[a] == nums[b]: c += 1
        return c


        

File: solutions/Question 3871/Solution2.cpp
/*
 * Problem: 3871. Count Commas in Range II
 * Difficulty: Medium
 * Topics: Math
 * URL: https://leetcode.com/problems/count-commas-in-range-ii/
 * Runtime: 0 ms
 * Memory: 8.9 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    long long countCommas(long long n) {
        long long count = 0;
        if (n >= 1000) count += n - 999;
        if (n >= 1000000LL) count += n - 999999LL;
        if (n >= 1000000000LL) count += n - 999999999LL;
        if (n >= 1000000000000LL) count += n - 999999999999LL;
        if (n >= 1000000000000000LL) count += n - 999999999999999LL;
        
        return count;

        
    }
};